service/src/main/java/tds/config/services/impl/ConfigServiceImpl.java

```diff
@@ -45,11 +45,17 @@ public ConfigServiceImpl(ConfigRepository configRepository,
     }
 
     @Override
-    public Optional<ClientSystemFlag> findClientSystemFlag(String clientName, String auditObject) {
+    public Optional<ClientSystemFlag> findClientSystemFlag(String clientName, String type) {
+        /*
+            The type is the name of the ClientSystemFlag.  Ideally the source of data for this
+            method will be an in-memory collection; ClientSystemFlag values rarely change, making them a
+            strong candidate for caching.
+        */
+
```

Comment:
The caching can be at the service layer or the repository layer, depending on what we what to set as our convention.  We mostly (not sure it was a hard rule) cached at the repository layer when we did our performance improvements which was what was done here.  If we kept it the same way, we would cache the result of the call to get the list of ClientSystemFlags so that the first time someone called this service method it would call the repository and get them all instead of only getting one at a time and caching that.

Therefore each call to the service gets the in-memory (or potentially Redis cached) list of objects and filters to get the one they need.  The original idea was that we knew there weren't 1000 rows that would come back and so we could save some DB queries by calling it once and getting all of them in cache to start.